src/config/GeneralConfig.php

```diff
@@ -5162,6 +5182,29 @@ public function preserveImageColorProfiles(bool $value = true): self
         return $this;
     }
 
+    /**
+     * When set to `true`, Craft will not allow relationship fields to be lazy loaded. If a relationship field is accessed lazily while
+     * set to `true` an exception will be thrown. When set to `false` relationships may be loaded lazily.
+     *
+     * Commonly you will set this based on your environment to disable lazy loading during development but allow it in production so that production
+     * doesn't error out if one slips through.
+     *
+     * ```php
+     * ->preventLazyLoading(true)
+     * ```
+     *
+     * @group System
+     * @param bool $value
+     * @return self
+     * @see $preventLazyLoading
+     * @since 4.3.0
+     */
+    public function preventLazyLoading(bool $value = false): self
```

Comment:
I'll wait on the team’s feedback regarding the broader concept, but wanted to flag this default value—to match other fluent methods, it ought to be `true` so that `->preventLazyLoading()` (without arguments) _enables_ the feature.

Fluent method argument defaults don't have to agree with initial property values!

Reply:
Good catch. This is resolved in https://github.com/craftcms/cms/pull/12035/commits/fff2caaa605e3e7a858e975d6da9e86b5aee645f.